internal-docs/rfcs/002-feed-clickhouse-migration.md

```diff
@@ -0,0 +1,61 @@
+# RFC: Session/Error Feeds ClickHouse Migration
+
+* Status: In Progress
+* Author: [@Vadman97](https://github.com/Vadman97)
+
+## Summary
+
+This RFC describes how we will migrate the Session/Error Feeds from OpenSearch to ClickHouse.
+We'll also discuss how we can entirely stop using OpenSearch.
+As part of this change, we will stop using InfluxDB for Error Group metrics.
+
+## Motivation
+
+See [PRD: Session/Error Feeds ClickHouse Migration](../prds/002-feed-clickhouse-migration.md).
+
+## Proposed Implementation
+
+The migration from OpenSearch to ClickHouse will happen in a few phases:
+1. Replace the Error feed queries with ClickHouse
+2. Replace the Error Group metrics with ClickHouse
+3. Replace the Session feed queries with ClickHouse
+4. Implement CommandBar improvements using ClickHouse
+
+## Database modeling
+
+As a starting point, the database schema can be similar to the current postgres schema for sessions and errors. 
+However, we'll need to benchmark the performance of different approaches for querying sessions and errors in ClickHouse. 
+Ie.
+1. having 3 separate tables matching what’s stored in Postgres and join them at query time
+2. denormalize fields and save a copy of their values with each session/error (as we do with OpenSearch)
+3. treat queryable session/error columns as fields too and denormalize with a format like (field_type, field_name, field_value, id), then look up the full session/error objects in postgres after getting ids from clickhouse
+
+Migrating data should be straightforward with a ClickHouse query as per [this blog](https://clickhouse.com/blog/migrating-data-between-clickhouse-postgres).
```

Comment:
There's actually a couple [strategies](https://clickhouse.com/blog/migrating-data-between-clickhouse-postgres) in here. When I looked at this, it seems to figure out how to populate a Clickhouse db but there wasn't good information on how to keep this information in sync (i.e. forward writes).

With OpenSearch, we keep data in sync through application code. Would we want the same approach here?

Reply:
correct, we'd want to do dual writes via application code until we achieve feature parity, at which point we would remove opensearch writes. i'll clarify that here.